### ISSUE TITLE
Android memory growth when loop="true" - expected behavior?

### ISSUE THREAD

**Issue by derrickrc** [airbnb/lottie-android#1141]
Hi Gabriel,

I think my question/issue is similar to this one, but perhaps more specific: https://github.com/airbnb/lottie-android/issues/746

I am developing a kiosk-like tablet Android app and was hoping to have a Lottie animation loop as an "attract screen" to draw in customers. I am using the below animations in different activities and in the Android profiler I see that both increment Memory Usage at about 0.1 mb every 4 seconds or so when loop="true" and I have observed it hit OOM after enough time has passed. 

https://lottiefiles.com/3832-liquid-loading (gone now, for some reason)
https://lottiefiles.com/3801-pointer

If I stop the animation after some number of repeats, it stops incrementing. Is it expected behavior in Java / Android to consume memory when an animation is looping? If so, is there some way to essentially have an infinite loop and not eventually cause an OOM? Should I force a GC every couple of minutes on the loop screen?

I am new to coding and Android development so apologies for the newbie question - thanks!

**Comment by derrickrc** [comment]
Actually I realized forcing a GC doesn't work since the activity never finishes. It looks like one workaround is writing a handler that restarts the welcome screen every couple of minutes and effectively destroys / restarts the activity (GC seems to kick in shortly after each time), but not sure if there's a better option. Thanks again!

**Comment by gpeal** [comment]
This will probably be fixed by https://github.com/airbnb/lottie-android/commit/ef330316af45ea383732ed9e0c183a8b171ca190.

Can you try 3.0.0-beta2?
If not, can you track memory allocations with the profiler to find the root cause in your case?

**Comment by gpeal** [comment]
Closing for now in case it is fixed by that that commit. You can reopen this if you need to.

**Comment by derrickrc** [comment]
Thanks Gabriel. It looks like Lottie 3.0.0-beta2 has a dependency on androidx.appcompat:appcompat:${androidXVersion}, right? I'm still on 28.0.0 and I got a lot of gradle build errors (I believe due to dependency conflicts) and tried excluding the androidx. files through packagingOptions - and then I got this error: 

Error: Program type already present: com.airbnb.lottie.animation.content.BaseStrokeContent$1

I tried messing with exclude group, but wasn't having luck. Is there any way to use 3.0.0-beta2 without migrating to androidx? Prior to this I was using Lottie 2.5.0.

Thanks again!
